pepr/attack_runner.py

```diff
@@ -0,0 +1,168 @@
+"""
+Attack runner for convenient pentesting with multiple configurations.
+"""
+
+import os
+import logging
+import pickle
+
+import numpy as np
+import tensorflow as tf
+import yaml
+
+from pepr import report
+from pepr.privacy import mia, gmia
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+def run_attacks(yaml_path, attack_obj_save_path, functions):
+    """
+    Run multiple attacks configured in a YAML file.
+
+    Parameters
+    ----------
+    yaml_path : str
+        Path to attack configuration file in YAML format.
+    attack_obj_save_path : str
+        Path where to save the serialized attack objects.
+    functions : dict
+        Dictionary containing the functions which return TensorFlow models. The keys
+        should correspond to the strings in the configuration file.
+
+    Returns
+    -------
+    dict
+        Dictionary with attack alias and the corresponding path to the serialized attack
+        object in the same order as specified in the YAML configuration.
+    """
+
+    def load_function_pointer(key, f_key) -> dict:
+        return {key: functions[f_key]}
+
+    def load_numpy(key, path) -> dict:
+        return {key: np.load(path)}
+
+    # Prefix to function mapping. If a key starts with one of these prefixes, the prefix
+    # gets removed and the function will be called with the value.
+    prefix_function_map = {
+        "<fn>": load_function_pointer,
+        "<np>": load_numpy,
+    }
+
+    # Attack type to attack constructor mapping. Add new attacks here.
+    attack_constructor_map = {
+        "mia": mia.Mia,
+        "gmia": gmia.DirectGmia,
+    }
```

Comment:
Probably we can also leave out this dict in future. However, for now that is fine and much better than before 👍